### PR TITLE
Apply for job wizard via modals and missing job info

### PR DESCRIPTION
**Pull request by bex1**
image

image

image

Should be deployed in sync with API since the create application route requirements has changed.

### CODE REVIEW

**Comment by bex1** [comment]
@buren The frilans finans consent on the request are not enforced by the API yet. We should make sure that it is enforced before deploying. Also the hints for each requested trait should likely be passed since its hard to make out what we really want when for example a specific skill or language is requested.

**Comment by buren** [comment]
@bex1 you're absolutely right, the code for enforcing it hasn't been merged yet. Here is the PR https://github.com/justarrived/just_match_api/pull/1040

You can track the progress of adding hints in this issue: https://github.com/justarrived/just_match_api/issues/1050

**Comment by buren** [comment]
On another note we actually don't need to gather the consent at this stage.. we could actually collect that when the users "confirms" the job/sets `will_perform`.. 😳

This API would look exactly the same as it is in justarrived/just_match_api#1040, however the update would move that to `POST /api/v1/jobs/:id/acceptances` instead. 

We'll look at it tomorrow :)

_EDIT_: On the other hand its not at all the end of the world if we collect the consent at an earlier stage. The important part is that the user performing the job has given their consent.

**Comment by buren** [comment]
cc @annahornmark, see the screenshots above for the new job application behaviour ☝️

**Comment by buren** [comment]
Potentially we could skip adding Frilans Finans logo, it might confuse users seeing another brand like that? On the other hand I like that its very clear/informative, but just wanted to highlight another perspective ;)

**Comment by bex1** [comment]
@buren  Yes we will discuss the consent tomorrow.

Regarding the frilans logo. I added it only to make the input field look a little better.  It looks rather weak with just the checkbox and link. 😄

File: src/app/components/maps/jobs-map/jobs-map.component.ts [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+        [zoom]="mapZoom"
+        class="jobs-map">
+        <sebm-google-map-marker
+          [latitude]="mapUserLocation.latitude"
+          [longitude]="mapUserLocation.longitude"
+          *ngIf="mapUserLocation">
+        </sebm-google-map-marker>
+        <job-map-marker
+          [job]="job"
+          *ngFor="let job of jobs | async">
```
Comment (buren) [inline src/app/components/maps/jobs-map/jobs-map.component.ts:35]: Still love `async` ❤️

File: src/app/components/maps/jobs-map/jobs-map.component.ts [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+
+  public jobs: Promise<Job[]>;
+  public mapError: string;
+  public mapErrorShow: boolean = false;
+  public mapLocation: MapLocation = this.defaultMapLocation;
+  public mapStyles = this.defaultMapStyle;
+  public mapUserLocation: MapLocation;
+  public mapZoom: number = this.defaultMapZoom;
+
+  constructor(
```
Comment (buren) [inline src/app/components/maps/jobs-map/jobs-map.component.ts:66]: Written a few Google Maps integrations.. none of them are this easy to follow 🙇

**Review by buren — APPROVED** [review]
As always: easy to follow and just the right amount of code needed to get the job done (and more!) 👍

**Comment by buren** [comment]
The only comment I got is that we should try to avoid clumping multiple unrelated PRs together. If we at some point would like to reject any of the changes it would be hard to keep the rest :)

So when possible make unrelated changes in another branch (I know you typically do this anyways and usually have good reasons when you not to, just a friendly reminder).

**Pull request merged**